cloud/mock.go

```diff
@@ -140,7 +142,35 @@ func (mockMgr *mockManager) SpawnHost(ctx context.Context, h *host.Host) (*host.
 	return h, nil
 }
 
-func (mockMgr *mockManager) ModifyHost(context.Context, *host.Host, host.HostModifyOptions) error {
+func (mockMgr *mockManager) ModifyHost(ctx context.Context, host *host.Host, changes host.HostModifyOptions) error {
+	l := mockMgr.mutex
+	l.Lock()
+	defer l.Unlock()
+	var err error
+	instance, ok := mockMgr.Instances[host.Id]
+	if !ok {
+		return errors.Errorf("unable to fetch host: %s", host.Id)
+	}
+	host.AddTags(changes.AddInstanceTags)
+	if err := host.SetTags(); err != nil {
+		return errors.Errorf("error adding tags in db")
+	}
+	instance.Tags = host.InstanceTags
+	mockMgr.Instances[host.Id] = instance
+
+	host.DeleteTags(changes.DeleteInstanceTags)
+	if err := host.SetTags(); err != nil {
+		return errors.Errorf("error deleting tags in db")
+	}
+	instance.Tags = host.InstanceTags
+	mockMgr.Instances[host.Id] = instance
+
+	if err = host.SetInstanceType(changes.InstanceType); err != nil {
```

Comment:
Should add/delete/setInstance only . be called if that change field is populated?